ui/app/AppLayouts/Wallet/popups/buy/BuyCryptoModal.qml

```diff
@@ -0,0 +1,141 @@
+                    root.buyCryptoAdaptor.fetchProviderUrl(
+                                root.buyCryptoInputParamsForm.selectedProviderId,
+                                buyCryptoProvidersListPanel.currentTabIndex,
+                                root.buyCryptoInputParamsForm.selectedWalletAddress,
+                                root.buyCryptoInputParamsForm.selectedNetworkChainId,
+                                root.buyCryptoAdaptor.selectedToken.symbol
+                                )
+                }
+            }
+            enabled: root.buyCryptoInputParamsForm.filledCorrectly
+        }
+    }
+
+    width: 560
+    height: 515
+    padding: Style.current.xlPadding
+    stackTitle: qsTr("Buy assets for %1").arg(!!buyCryptoAdaptor.selectedAccount ? buyCryptoAdaptor.selectedAccount.name: "")
+    rightButtons: [d.buyButton, finishButton]
+    finishButton: StatusButton {
+        text: qsTr("Done")
+        onClicked: root.close()
+    }
+
+    Connections {
+        target: root.buyCryptoAdaptor
+        function onProviderUrlReady(url) {
+            if (!!root.buyCryptoAdaptor.selectedProvider && !!url)
+                Global.openLinkWithConfirmation(url, root.buyCryptoAdaptor.selectedProvider.hostname)
+            root.close()
+        }
+    }
+
+    onOpened: root.buyCryptoAdaptor.fetchProviders()
+    onClosed: {
+        // reset the view
+        root.replaceItem = undefined
+        buyCryptoProvidersListPanel.currentTabIndex = 0
+        root.buyCryptoAdaptor.reset()
+        root.buyCryptoInputParamsForm.resetFormData()
+    }
+
+    stackItems: [
+        BuyCryptoProvidersListPanel {
+            id: buyCryptoProvidersListPanel
+            providersLoading: root.buyCryptoAdaptor.providersLoading
+            providersModel: root.buyCryptoAdaptor.providersModel
+            selectedProviderId: root.buyCryptoInputParamsForm.selectedProviderId
+            isUrlBeingFetched: root.buyCryptoAdaptor.urlIsBeingFetched
+            onProviderSelected: {
+                root.buyCryptoInputParamsForm.selectedProviderId = model.id
+                if(model.urlsNeedParameters) {
+                    root.replace(selectParamsPanel)
+                } else {
+                    root.buyCryptoAdaptor.fetchProviderUrl(model.id, currentTabIndex)
+                }
+            }
+        }
+    ]
+
+    Component {
+        id: selectParamsPanel
+        SelectParamsForBuyCryptoPanel {
+            id: selectParamsPanelInst
+            adaptor: TokenSelectorViewAdaptor {
+                /* TODO these should be hadbled and perhaps improved under
+                https://github.com/status-im/status-desktop/issues/16025 */
+                assetsModel: SortFilterProxyModel {
+                    sourceModel: root.buyCryptoAdaptor.groupedAccountAssetsModelWithKey
+                    filters: FastExpressionFilter {
+                        expression: model.addressPerChain.rowCount() > 0
```

Comment:
`rowCount()` shouldn't really be used as it's static, w/o a notifier signal

Reply:
model.addressPerChain.ModelCount.count doesnt work! idk why @caybro

Reply:
Hmm.... :thinking: maybe because of the way `FastExpressionFilter` exposes the roles? CC @micieslak

Reply:
It is that's similar story as with singletons. The expression uses child context of the script's context (as there are role values injected). However child context doesn't inherit imports. It causes that both singletons and attached properties are not accessible there. If the submodel is not intended to be changed, using `rowCount()` is maybe not ideal but should work ok.